### ISSUE TITLE
not compatible with typescript "strict": true

### ISSUE THREAD

**Issue by aligator** [ticruz38/graphql-codegen-svelte-apollo#22]
Hi,

I just noticed that the generated code breaks when I set "strict" to true:

Out of 
```graphql
query GetNode($path: String!) {
    getNode(path: $path) {
        id,
        name,
        description,
        isFolder,
        mimeType,
        children {
            id,
            name,
            description,
            isFolder,
            mimeType
        }
    }
}
```

I get 
```ts
export const GetNode = (
            options: Omit<
              WatchQueryOptions<GetNodeQueryVariables>, 
              "query"
            >
          ): Readable<
            ApolloQueryResult<GetNodeQuery> & {
              query: ObservableQuery<
                GetNodeQuery,
                GetNodeQueryVariables
              >;
            }
          > => {
            const q = client.watchQuery({
              query: GetNodeDoc,
              ...options,
            });
            var result = readable<
              ApolloQueryResult<GetNodeQuery> & {
                query: ObservableQuery<
                  GetNodeQuery,
                  GetNodeQueryVariables
                >;
              }
            >(
              { data: null, loading: true, error: null, networkStatus: 1, query: null },
              (set) => {
                q.subscribe((v) => {
                  set({ ...v, query: q });
                });
              }
            );
            return result;
          }
        export const GetNode = (
            options: Omit<
              WatchQueryOptions<GetNodeQueryVariables>, 
              "query"
            >
          ): Readable<
            ApolloQueryResult<GetNodeQuery> & {
              query: ObservableQuery<
                GetNodeQuery,
                GetNodeQueryVariables
              >;
            }
          > => {
            const q = client.watchQuery({
              query: GetNodeDoc,
              ...options,
            });
            var result = readable<
              ApolloQueryResult<GetNodeQuery> & {
                query: ObservableQuery<
                  GetNodeQuery,
                  GetNodeQueryVariables
                >;
              }
            >(
              { data: null, loading: true, error: null, networkStatus: 1, query: null },
              (set) => {
                q.subscribe((v) => {
                  set({ ...v, query: q });
                });
              }
            );
            return result;
          }
```

and the line 
```
              { data: null, loading: true, error: null, networkStatus: 1, query: null },
```

is wrong:
```
(!) Plugin typescript: @rollup/plugin-typescript TS2322: Type 'undefined' is not assignable to type 'GetNodeQuery'.
ui/src/api/types.ts: (110:17)

110               { data: null, loading: true, error: null, networkStatus: 1, query: null },
                    ~~~~

(!) Plugin typescript: @rollup/plugin-typescript TS2322: Type 'null' is not assignable to type 'ApolloError | undefined'.
ui/src/api/types.ts: (110:49)

110               { data: null, loading: true, error: null, networkStatus: 1, query: null },
                                                    ~~~~~

(!) Plugin typescript: @rollup/plugin-typescript TS2322: Type 'null' is not assignable to type 'ObservableQuery<GetNodeQuery, Exact<{ path: string; }>>'.
ui/src/api/types.ts: (110:80)

110               { data: null, loading: true, error: null, networkStatus: 1, query: null },

```

Maybe a possible solution would be to just set the value to __undefined__ directly instead of this whole `{ data: null, loading: true, error: null, networkStatus: 1, query: null }`?
To avoid breaking changes this could be implemented with a generation flag such as `tsStrictCompatible: true`

**Comment by ticruz38** [comment]
Hmmm no setting the result to undefined would be painful to develop with, I'll see how I could fix this, there should be a class ApolloResult or smth similar to instanciate with empty params.

**Comment by ticruz38** [comment]
also, you should put the generated files to your exclude list in tsconfig, you don't need to typecheck auto-generated files, this will just slow down your development workflow.

**Comment by ticruz38** [comment]
I have published a new version 1.0.1 that should fix your problem, let me know how it goes

**Comment by aligator** [comment]
Thanks :-)
